yesod-form/Yesod/Form/Fields.hs

```diff
@@ -948,3 +951,23 @@ prependZero t0 = if T.null t1
 -- The basic datastructure used is an 'Option', which combines a user-facing display value, the internal Haskell value being selected, and an external 'Text' stored as the @value@ in the form (used to map back to the internal value). A list of these, together with a function mapping from an external value back to a Haskell value, form an 'OptionList', which several of these functions take as an argument.
 -- 
 -- Typically, you won't need to create an 'OptionList' directly and can instead make one with functions like 'optionsPairs' or 'optionsEnum'. Alternatively, you can use functions like 'selectFieldList', which use their @[(msg, a)]@ parameter to create an 'OptionList' themselves.
+
+-- | Creates an input with @type="color"@.
+--   The input value must be provided in hexadecimal format #rrggbb.
+--
+-- @since 1.7.1 
+colorField :: Monad m => RenderMessage (HandlerSite m) FormMessage => Field m Text
+colorField = Field
+    { fieldParse = parseHelper $ \s ->
+        if isHexColor $ unpack s then Right s
+        else Left $ MsgInvalidHexColorFormat s
+    , fieldView = \theId name attrs val _ -> [whamlet|
+$newline never
+<input ##{theId} name=#{name} *{attrs} type=color value=#{either id id val}>
+|]
+    , fieldEnctype = UrlEncoded
+    }
+  where
+      isHexColor :: String -> Bool
+      isHexColor ['#',a,b,c,d,e,f] = all isHexDigit [a,b,c,d,e,f]
+      isHexColor _ = False
```

Comment:
Hex numbers in CSS can also be `#fff` (3 letters). Are you sure that all browsers ever only set 6 letter hex digits?

Reply:
Quoting from [https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/color](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/color): 

> "The value must be in seven-character hexadecimal notation, meaning the "#" character followed by two digits each representing red, green, and blue, like this: #rrggbb".

It is supposed to cover only HTML, not CSS.
Please let me know if I have to look in other non-HTML specs such as CSS. I'm not sure of this.

Reply:
Ah, I guess you're right! BTW, this would be the most official spec for HTML input: https://html.spec.whatwg.org/multipage/input.html#color-state-(type=color)
… the "bookkeeping details" are also interesting.